File: python/mirage/vfs/lancedb/prompt.py
# ========= Copyright 2026 @ Strukto.AI All Rights Reserved. =========
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ========= Copyright 2026 @ Strukto.AI All Rights Reserved. =========

PROMPT = """\
This mount is a LanceDB table exposed as a filesystem.

Layout:
- Directories are the configured group-by columns. Descending narrows a filter,
  e.g. `ls Men/Tshirts/Blue` lists rows where gender=Men, articleType=Tshirts,
  baseColour=Blue.
- Each matching row appears as two files: `<id>.md` (a readable card with all
  attributes) and `<id>.<ext>` (the raw blob/image bytes), when a blob column is
  configured.
- Semantic search is a virtual folder named `_search`. Read a query as a path
  segment: `ls "_search/red running shoes"` returns the top matches as row
  files; `cat "_search/red running shoes/<id>.md"` shows the card with a score.

Use ls/cd/cat/tree/find/grep as usual. Quote queries that contain spaces.\
"""

File: typescript/packages/core/src/vfs/postgres/prompt.ts
// ========= Copyright 2026 @ Strukto.AI All Rights Reserved. =========
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ========= Copyright 2026 @ Strukto.AI All Rights Reserved. =========

export const POSTGRES_PROMPT = `{prefix}
  database.json                  cross-schema graph + sizes
  <schema>/                      Postgres schema (namespace)
    tables/<table>/
      schema.json                column types, PK/FK, indexes
      rows.jsonl                 data (size-guarded)
    views/<view>/
      schema.json
      rows.jsonl
  Read database.json first to plan joins. Reading rows.jsonl is refused
  for tables above the configured row/byte threshold; use head, tail, wc,
  or grep, all of which push predicates down to SQL.`
